### PR TITLE
Use compose theming engine, (drop compatibility MdcTheming etc)

### LINKED ISSUE #47: Use compose theming engine, (drop compatibility MdcTheming etc)
- [x] Light theme
- [x] Dark theme
- [ ] Get rid of Accompanist dependency
- [x] Drop styles from XML (all migrations must be done before)
- [x] Migrate night mode forcing to Compose.

Depends on #41
Depends on #49
Depends on #50

### PR DESCRIPTION
**Pull request by ArnyminerZ**

### CODE REVIEW

File: app/src/main/java/at/bitfire/icsdroid/ui/AddCalendarActivity.kt [hunk 1]
```diff
@@ -111,132 +113,128 @@ class AddCalendarActivity : AppCompatActivity() {
             message?.let { Toast.makeText(this, it, Toast.LENGTH_LONG).show() }
         }
 
-        setContent {
-            MdcTheme {
-                val pagerState = rememberPagerState { 2 }
+        setContentThemed {
+            val scope = rememberCoroutineScope()
```
Comment (sunkup) [inline app/src/main/java/at/bitfire/icsdroid/ui/AddCalendarActivity.kt]: Why add the scope val here? I can't see any usages :thinking:

File: app/src/main/java/at/bitfire/icsdroid/ui/theme/Colors.kt
Comment (sunkup) [inline app/src/main/java/at/bitfire/icsdroid/ui/theme/Colors.kt:1]: Why are there so many new unused colors? The old `styles.xml` file had only about eight?

File: app/src/main/java/at/bitfire/icsdroid/ui/theme/Theme.kt [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+/**
+ * Set the Jetpack Compose UI content for this view. Initial composition will occur when the view
+ * becomes attached to a window or when `createComposition` is called, whichever comes first.
+ *
+ * Then, applies [AppTheme] to the UI.
+ *
+ * @param content A `@Composable` function declaring the UI contents
+ */
+@Deprecated("Fragments should not be used")
+fun ComposeView.setContentThemed(content: @Composable () -> Unit) {
```
Comment (sunkup) [inline app/src/main/java/at/bitfire/icsdroid/ui/theme/Theme.kt]: I guess this is obsolete now? I can't find any usages.

File: app/src/main/res/values/colors.xml
Comment (sunkup) [inline app/src/main/res/values/colors.xml:1]: Why does this color need its own file?

**Review by sunkup — CHANGES_REQUESTED** [review]
Seems to work alright, but see the comments.

And then there is the system bars, which in light mode should show light buttons as well as in dark mode, no? It looks wrong with dark buttons in light mode like this at least ... 

<details>

<summary>Pics</summary>

Screenshot from 2024-02-12 10-36-20

Screenshot from 2024-02-12 10-36-43


</details>

File: app/src/main/java/at/bitfire/icsdroid/ui/AddCalendarActivity.kt (hunk 1 above)
Comment (ArnyminerZ) [reply app/src/main/java/at/bitfire/icsdroid/ui/AddCalendarActivity.kt]: Maybe it got lost at some point during merging with dev, I'll remove it

File: app/src/main/java/at/bitfire/icsdroid/ui/theme/Colors.kt
Comment (ArnyminerZ) [reply app/src/main/java/at/bitfire/icsdroid/ui/theme/Colors.kt:1]: We were leaving a lot of colors to the value by default. Actually I believe we can do the same here. I'll just leave the colors we defined, and see how the theme changes

File: app/src/main/res/values/colors.xml
Comment (ArnyminerZ) [reply app/src/main/res/values/colors.xml:1]: It's used for the launcher icon

Comment (ArnyminerZ) [reply app/src/main/res/values/colors.xml:1]: https://github.com/ArnyminerZ/icsx5/blob/8a422db6253518ddfb60113a92498daef0fed009/app/src/main/res/mipmap-anydpi-v26/ic_launcher.xml#L2

**Review by sunkup — CHANGES_REQUESTED** [review]
Nice. 

I don't like the dark theme, but I guess it's not important at the moment :)

Can remove the  Accompanist dependency now?

EDIT: Ah, I can see you do that in the Migration to Gradle Version Catalog :+1:

**Review by sunkup — APPROVED** [review]

**Pull request merged**